yaksh/code_evaluator.py

```diff
@@ -97,45 +113,55 @@ def setup(self):
         if self.in_dir:
             if not os.path.exists(self.in_dir):
                 os.makedirs(self.in_dir)
-        self._change_dir(self.in_dir)
 
-    def safe_evaluate(self, user_answer, partial_grading, test_case_data, file_paths=None):
+    def get_evaluator_objects(self, kwargs):
+        metadata = kwargs.get('metadata')
+        test_case_data = kwargs.get('test_case_data')
```

Comment:
I had earlier assumed that we may need to pop some data or may have to pass the arguments to different classes/methods (metadata is mostly question/user answer related and test case data is test case specific)

I guess I also assumed that this would be more readable and segregated.